.github/workflows/ci-cd.yml

```diff
@@ -0,0 +1,203 @@
+
+      - if: steps.conditionals_handler.outputs.is_push_to_default_branch == 'true'
+        name: Project version/deploy determiner
+        id: proj_ver_determiner
+        shell: sh
+        run: |
+          git fetch --all --tags
+
+          PACKAGE_VERSION=$(cat package.json \
+            | grep version \
+            | head -1 \
+            | awk -F: '{ print $2 }' \
+            | sed 's/[",]//g')
+          PACKAGE_VERSION_TAG=`echo v$PACKAGE_VERSION | sed -e 's/[[:space:]]//'`
+          LATEST_GITHUB_TAG=`echo $(git tag | sort --version-sort | tail -n1)`
+          SHOULD_DEPLOY="false"
+
+          # download semver compare tool
+          curl https://raw.githubusercontent.com/Ariel-Rodriguez/sh-semversion-2/main/semver2.sh -o semver2.sh
+          chmod +x semver2.sh
+
+          if [ `echo $(./semver2.sh $PACKAGE_VERSION_TAG $LATEST_GITHUB_TAG)` = 1 ]; then
+            SHOULD_DEPLOY="true"
+          fi
+
+          echo "::set-output name=package_version::$(echo $PACKAGE_VERSION)"
+          echo "::set-output name=package_version_tag::$(echo $PACKAGE_VERSION_TAG)"
+          echo "::set-output name=latest_github_tag::$(echo $LATEST_GITHUB_TAG)"
+          echo "::set-output name=should_deploy::$(echo $SHOULD_DEPLOY)"
+
+      # Build and test validation
+
+      - name: Create logs directory
+        shell: sh
+        run: |
+          mkdir -p logs
+
+      - name: Install dependencies
+        shell: sh
+        run: |
+          npm ci --loglevel verbose 2>&1 | tee logs/install.log
+
+      - name: Build the app
+        shell: sh
+        run: |
+          npm run build --if-present --loglevel verbose 2>&1 | tee logs/build.log
+        env:
+          TEST_NO_SANDBOX: 1
+          TERSER_TEST_ALL: 1
+
+      - if: matrix.node-version != '10.x'
+        name: Run compress tests
+        shell: sh
+        run: |
+          npm run test:compress --loglevel verbose 2>&1 | tee logs/compress_test.log
+
+      - if: matrix.node-version != '10.x'
+        name: Run mocha tests
+        shell: sh
+        run: |
+          npm run test:mocha --loglevel verbose 2>&1 | tee logs/mocha_test.log
+
+      - if: matrix.node-version == '10.x'
+        name: Run all tests with Node 10.x
+        shell: sh
+        run: |
+          node --require esm test/compress.js --loglevel verbose 2>&1 | tee logs/compress_test.log
+          npm run test:mocha -- --require esm --loglevel verbose 2>&1 | tee logs/mocha_test.log
+        env:
+          TEST_NO_SANDBOX: 1
+
+      - if: matrix.os == 'ubuntu-latest' && matrix.node-version == '12.x'
+        name: Run functional tests on ${{ matrix.os }} with Node 12.x
+        run: |
+          ./test/functional.sh 2>&1 | tee logs/functional_tests.log
+
+      # Artifact publish to pipeline
+
+      - name: Upload logs as artifact
+        uses: actions/upload-artifact@v2
+        with:
+          name: "Logs ${{ matrix.os }} - nodejs ${{ matrix.node-version }}"
+          path: logs/
+
+  deploy:
```

Comment:
This entire step should be left out, I believe.

Terser is still published to NPM by hand, and moving to a flow like that should at least be a separate PR

Reply:
I think there shouldn't be a `deploy` step for now.

Reply:
Understood.

We'll make the changes promptly 🙂